instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/SpanKey.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+// TODO make class (and fields/methods below) package-private after tracers are gone and no need for
+//  bridging from ClientSpan/ServerSpan/ConsumerSpan
+public final class SpanKey {
+
+  private static final ContextKey<Span> SERVER_KEY =
+      ContextKey.named("opentelemetry-traces-span-key-server");
+  private static final ContextKey<Span> CONSUMER_KEY =
+      ContextKey.named("opentelemetry-traces-span-key-consumer");
+
+  // TODO bridge these constants in AgentContextStorage
+  private static final ContextKey<Span> HTTP_KEY =
+      ContextKey.named("opentelemetry-traces-span-key-http");
+  private static final ContextKey<Span> RPC_KEY =
+      ContextKey.named("opentelemetry-traces-span-key-rpc");
+  private static final ContextKey<Span> DB_KEY =
+      ContextKey.named("opentelemetry-traces-span-key-db");
+  private static final ContextKey<Span> MESSAGING_KEY =
+      ContextKey.named("opentelemetry-traces-span-key-messaging");
+
+  // this is used instead of above, depending on the configuration value for
+  // otel.instrumentation.experimental.span-suppression-by-type
+  // named outgoing because it covers both client and producer spans
+  private static final ContextKey<Span> OUTGOING_KEY =
+      ContextKey.named("opentelemetry-traces-span-key-client");
+
+  public static final SpanKey SERVER = new SpanKey(SERVER_KEY);
+  public static final SpanKey CONSUMER = new SpanKey(CONSUMER_KEY);
+
+  static final SpanKey HTTP = new SpanKey(HTTP_KEY);
+  static final SpanKey RPC = new SpanKey(RPC_KEY);
+  static final SpanKey DB = new SpanKey(DB_KEY);
+  static final SpanKey MESSAGING = new SpanKey(MESSAGING_KEY);
```

Comment:
We could name that one `PRODUCER` - the `CONSUMER` is also messaging related after all

Reply:
this is the same for HTTP and RPC - they can be both server and client, but those are client only. May be name should reflect that? `HTTP_CLIENT`, `MESSAGING_PRODUCER`?

Reply:
this makes sense to me 👍